### PR TITLE
ENH: add random_state param in dim reduction methods

### PR DESCRIPTION
**Pull request by misialq**
This PR introduces a `random_state` parameter to the two dimensionality reduction methods: tSNE and UMAP for better result reproducibility.

### CODE REVIEW

File: q2_diversity/_ordination.py [hunk 1]
```diff
@@ -38,15 +38,17 @@ def tsne(distance_matrix: skbio.DistanceMatrix,
          perplexity: float = 25.0,
          n_iter: int = 1000,
          learning_rate: float = 200.0,
-         early_exaggeration: float = 12.0) -> skbio.OrdinationResults:
+         early_exaggeration: float = 12.0,
+         random_state: int = 1) -> skbio.OrdinationResults:
```
Comment (ebolyen) [inline q2_diversity/_ordination.py:42]: This should default to `None` so that it's optional (it's also a valid input for sklearn with the expected semantics).

File: q2_diversity/_ordination.py [hunk 2]
```diff
@@ -68,14 +70,16 @@ def tsne(distance_matrix: skbio.DistanceMatrix,
 def umap(distance_matrix: skbio.DistanceMatrix,
          number_of_dimensions: int = 2,
          n_neighbors: int = 15,
-         min_dist: float = 0.4) -> skbio.OrdinationResults:
+         min_dist: float = 0.4,
+         random_state: int = 1) -> skbio.OrdinationResults:
```
Comment (ebolyen) [inline q2_diversity/_ordination.py:74]: Same here (umap package also supports `None`).

**Review by ebolyen — CHANGES_REQUESTED** [review]

**Comment by misialq** [comment]
Thanks for pointing those out @ebolyen, all's updated!

**Review by ebolyen — APPROVED** [review]
Thanks @misialq!
